core/src/replay_stage.rs

```diff
@@ -1829,9 +1829,12 @@ impl ReplayStage {
                 } else if let Some(prev_hash) =
                     duplicate_confirmed_slots.insert(confirmed_slot, duplicate_confirmed_hash)
                 {
-                    assert_eq!(prev_hash, duplicate_confirmed_hash);
+                    assert_eq!(
+                        prev_hash, duplicate_confirmed_hash,
+                        "Additional duplicate confirmed notification with a different hash"
+                    );
```

Comment:
nit: I could see the slot being useful here, maybe the hashes too ?

Reply:
Added the slot, we get the hashes for free from the assertion:
```
assertion `left == right` failed: Additional duplicate confirmed notification for 6 with a different hash
  left: AYvBWrMJ9gDkQa23LxbVRFuthLeiLzHPz5CPRNV5b3gV
 right: 8tEmgfKVZTiwRA6poeK5xJJ2zZCSc9z2ne2qK5rHEMM
 ```

Reply:
Ahh right!